### PR TITLE
perf(wallet): use vanilla js for nft iframe

### PR DESCRIPTION
**Pull request by josheleonard**
Resolves https://github.com/brave/brave-browser/issues/39024

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
NFT images should continue to load without regressions

### CODE REVIEW

File: components/brave_wallet_ui/nft/nft_script.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+function render() {
+  const imgEl = document.getElementById('nft-image') as HTMLImageElement
+  const imgWrapperEl = document.getElementById('nft-image-wrapper')!
+
+  // unhide elements
+  imgWrapperEl.className = 'image-wrapper'
+  imgEl.className = 'image'
+
+  // update img src
+  imgEl.src = displayMode === 'details' && nftMetadata ? mediaUrl : imageSrc
```
[the remark this answers is not in the document]
Comment (diracdeltas) [reply components/brave_wallet_ui/nft/nft_script.ts:46]: This is probably a legit alert. I think `window.location.search` can be overwritten by the page, which means mediaUrl is untrusted input that could be `javascript:...` for instance.

[the remark this answers is not in the document]
Comment (diracdeltas) [reply components/brave_wallet_ui/nft/nft_script.ts:46]: Actually I think this is running in `chrome-untrusted://nft-display?...` which should be fine unless the NFT itself can change the location of this page somehow. Can you remind me how this works again? Is chrome://wallet iframing chrome-untrusted://nft-display which then iframes the NFT? Are there any sandbox restrictions on the innermost NFT?

[the remark this answers is not in the document]
Comment (josheleonard) [reply components/brave_wallet_ui/nft/nft_script.ts:40]: Only one layer of iframing. `chrome://wallet` loads an iframe of `chrome-untrusted://nft-display`, which loads the image provided within `window.location.search` into a `img` tag.

[the remark this answers is not in the document]
Comment (josheleonard) [reply components/brave_wallet_ui/nft/nft_script.ts:40]: For sandboxing we currently use both `allow-scripts` and `allow-same-origin` for some iframes. `allow-same-origin` is needed prevent breakage when `IsolateSandboxedIframes` is enabled (https://chromium-review.googlesource.com/c/chromium/src/+/5538608) (https://bravesoftware.slack.com/archives/CU5UXM3EC/p1718652484089379?thread_ts=1717054621.009079&cid=CU5UXM3EC)

[the remark this answers is not in the document]
Comment (diracdeltas) [reply components/brave_wallet_ui/nft/nft_script.ts:40]: ok, seems fine and @thypon says he vetted this approach. thanks

[the remark this answers is not in the document]
Comment (josheleonard) [reply components/brave_wallet_ui/nft/nft_script.ts:40]: @diracdeltas, should I create a security review issue for this PR?

[the remark this answers is not in the document]
Comment (diracdeltas) [reply components/brave_wallet_ui/nft/nft_script.ts:40]: i think it's ok

**Review by muliswilliam — APPROVED** [review]
LGTM

**Pull request merged**